codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -1060,8 +1038,8 @@ void removeFromActiveStreams(DefaultStream stream) {
                         listeners.get(i).onStreamClosed(stream);
                     }
                 } finally {
-                    // Mark this stream for removal.
-                    removalPolicy.markForRemoval(stream);
+                    // Remove the stream entirely if it has no children in the priority tree.
```

Comment:
> @buchgr I believe the comment accurately reflects the current behavior
+1.

The current behavior is such that streams remain in the priority tree until `prioritizableForTree() == 0`.  Events which can change this count (parent change, stream removal from tree, state change, etc...) re-evaluate whether the stream still belongs in the tree, and calls `removeStream` if not. I'm not sure we need this comment here though...`removeStream` has a pretty detailed comment on it...maybe just clean that comment up if it is not adequite?